### PR TITLE
Add async tests for plugins

### PR DESCRIPTION
**Pull request by florianvazelle**
### Description
1. Add asynchronous testing for plugins
2. Clean title in USER GUIDE for alerting section 

### Issues Resolved
1. To increase coverage
2. As requested here https://github.com/opensearch-project/opensearch-py/pull/398#discussion_r1245755900

### CODE REVIEW

**Comment by dblock** [comment]
LGTM, finish it up?

**Comment by florianvazelle** [comment]
Hi @dblock, there are two points I'd like to look at:
- the tests pass but the teardown of the last test is flaky, because the event loop seems closed
- and coverage isn't increasing, because it seems that coverage reports sent to codecov are from the test workflow (https://github.com/opensearch-project/opensearch-py/blob/main/.github/workflows/test.yml), and in the test workflow no OpenSearch is run. So in fact coverage doesn't really represent what's being tested. I'm taking a little more time to see if it isn't possible to have a workflow that would launch all the tests (with a running OpenSearch), and that would send a complete report, which would make it possible to have a real coverage report on plugins for example.

**Comment by dblock** [comment]
@florianvazelle I wouldn't worry about coverage

**Comment by florianvazelle** [comment]
@dblock Ok, so It's good to me now

File: test_opensearchpy/test_async/test_server/test_plugins/test_alerting.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
```
Comment (dblock) [inline test_opensearchpy/test_async/test_server/test_plugins/test_alerting.py:8]: You can remove the modifications part from new files. That was added during the fork for files that had an ES license before.

File: test_opensearchpy/test_async/test_server/test_plugins/__init__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
```
Comment (dblock) [inline test_opensearchpy/test_async/test_server/test_plugins/__init__.py:7]: This is a new file, so I believe everything including and down from Modifications ... should be removed. Just leave the SPDX header and the paragraph that says "The OpenSearch ... open source license.".

**Review by dblock — APPROVED** [review]
LGTM, two small changes with the license headers please? Sorry for the hassle.

tl;dr:

- new files only need the SPDX header + first paragraph
- existing files license headers should not be changed

File: test_opensearchpy/test_async/test_server/test_plugins/test_alerting.py (hunk 1 above)
Comment (florianvazelle) [reply test_opensearchpy/test_async/test_server/test_plugins/test_alerting.py:8]: But the `utils/license-headers.py` enforce the `Modifications Copyright OpenSearch Contributors` part. Should I adapt checks made in the script ?

Comment (dblock) [reply test_opensearchpy/test_async/test_server/test_plugins/test_alerting.py:8]: Hmmm, we can deal with this separately.
